crypto/crypto_aes.c

```diff
@@ -79,6 +83,13 @@ hwtest(struct aes_test * knowngood)
 	crypto_aes_encrypt_block_aesni(knowngood->ptext, ctext_hw, kexp_hw);
 	crypto_aes_key_free_aesni(kexp_hw);
 #endif
+#if defined(CPUSUPPORT_ARM_AES)
```

Comment:
I'm not incredibly happy with this -- I know we're never going to have CPUSUPPORT_X86_AESNI and CPUSUPPORT_ARM_AES defined at the same time, but this code looks like it's writing to `ctext_hw` twice.  Or zero times (again, also not possible since we would never call hwtest if neither of those is defined, but still...).

Would this work better as `hwtest_x86_aesni` and `hwtest_arm_aes` functions instead?

Reply:
Yeah, I winced at that a few times.  I have code that basically does that (via a callback to `hwtest()`), but I was saving that PR for after this one.  I'll swap the order and do that one first.

Reply:
Followed up in #405.